lib/internal/modules/esm/loader.js

```diff
@@ -260,12 +363,20 @@ class DefaultModuleLoader {
    */
   async load(url, context) {
     defaultLoad ??= require('internal/modules/esm/load').defaultLoad;
```

Comment:
Because this way they’re not loaded unless they’re needed/used.

Reply:
Is there performance implications for this? Should more things be loaded on-demand if I come across such a situation?

Reply:
We load at the top of the file if something is _always_ used; there’s no point in loading lazily if it happens every time. Sometimes things are loaded lazily because of cycles (the order of `require` calls getting resolved).

But there are plenty of cases where the default path doesn’t need a particular import, and so those get loaded lazily. We’re assuming that the default case for most Node apps is that no custom loaders are present, so most things related to custom loaders should be lazy if possible.

Reply:
Ah great thanks for clarifying! 🙏

Reply:
I think there is another condition where we delay the call to `require`: to control the sequence modules are loaded in order to speed up bootstraping. You'll see some paths that will always happen, but their dependency is delayed until that is actually hit, meaning earlier things happen faster.

I'm not an expert on this though—Antoine or Joyee are far more knowledgeable on it.